pandas/core/groupby.py

```diff
@@ -38,6 +38,23 @@
 import pandas.algos as _algos
 import pandas.hashtable as _hash
 
+
+class _ErrorContainer(Exception):
+    """Used internally to encapsulate an error. If need be, it should be caught
```

Comment:
This _is_ part of the solution to the current issue. I'm explaining that in the main discussion.